sapinvoices/sap.py

```diff
@@ -630,18 +631,22 @@ def mark_invoices_paid(
         logger.debug("date: %s", date)
         logger.debug("total amount: %s", invoice["total amount"])
         logger.debug("currency: %s", invoice["currency"])
-        response = alma_client.mark_invoice_paid(
-            invoice_id, date, invoice["total amount"], invoice["currency"]
-        )
-        if response["payment"]["payment_status"]["value"] == "PAID":
-            logger.debug("Invoice '%s' marked as paid in Alma", invoice_id)
-            paid_invoice_count += 1
-        else:
+        try:
+            response = alma_client.mark_invoice_paid(
+                invoice_id, date, invoice["total amount"], invoice["currency"]
+            )
+            if response["payment"]["payment_status"]["value"] == "PAID":
+                logger.debug("Invoice '%s' marked as paid in Alma", invoice_id)
+                paid_invoice_count += 1
+            else:
+                raise ValueError
```

Comment:
Do we need to test for a `ValueError` being raised as well?

Reply:
I think so, the idea is that we may get a valid response from the Alma API, but the payment status value is not equal to "PAID", and if it isn't PAID, we need to treat it the same way as if a request exception occurred, e.g. logging an error.